### PR TITLE
Update Networking API Specs for NRP

### PR DESCRIPTION
**Pull request by zhiyuanxia**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by zhiyuanxia** [comment]
@bganapa , @deathly809 , @martabarillas 
I am submitting a PR to fix the api scan issues in the NRP repo. 

This scan seems to be failing. public.rest-api-specs (lintdiff). Looking at the details, there doesn't seem to be any additional information. Do you know why it is failing?

**Review by bganapa — APPROVED** [review]

**Comment by bganapa** [comment]
@weidongxu-microsoft Could you please help why the CI gate is failing? There is not much details in the job links. Thanks!

File: specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/preview/2015-06-15/Network.json [hunk 1]
```diff
@@ -75,6 +75,98 @@
...
+        },
+        "description": "Returns the list of operation results for a location",
+        "tags": [
+          "Network"
+        ],
+        "operationId": "LocationsOperationResults_List",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter",
+            "$ref": "#/parameters/LocationParameter"
```
Comment (weidongxu-microsoft) [inline specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/preview/2015-06-15/Network.json]: I think lint report error on this line. You seem to have 2 ref in same object. I guess it should be in 2 objects.

**Comment by weidongxu-microsoft** [comment]
You can click the pipeline error for detailed message.

> abc831203@TSDM'[{\n "Channel": "error",\n "Text": "Syntax error: duplicate key",\n "Source": [\n {\n "document": "file:///home/vsts/work/1/s/specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/preview/2015-06-15/Network.json",\n "Position": {\n "line": 124,\n "column": 12\n }\n }\n ],\n "Range": [\n {\n "document":

**Comment by weidongxu-microsoft** [comment]
You can click the pipeline error for detailed message.

> abc831203@TSDM'[{\n "Channel": "error",\n "Text": "Syntax error: duplicate key",\n "Source": [\n {\n "document": "file:///home/vsts/work/1/s/specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/preview/2015-06-15/Network.json",\n "Position": {\n "line": 124,\n "column": 12\n }\n }\n ],\n "Range": [\n {\n "document":

**Comment by yungezz** [comment]
@zhiyuanxia  could you pls rename the PR title to a meaningful title?

File: specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/preview/2015-06-15/Network.json (hunk 1 above)
Comment (zhiyuanxia) [reply specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/preview/2015-06-15/Network.json]: Thank you! I updated the PR with my new changes and added a more meaningful title to the commit.

**Comment by zhiyuanxia** [comment]
@yungezz I updated my commit and title with a more meaningful title

**Comment by zhiyuanxia** [comment]
@weidongxu-microsoft I updated the changes so that all of the checks are now passing. Could you take a look to see if we can complete the PR? 

Thank you.

**Comment by zhiyuanxia** [comment]
Thank you Weidong.

 I updated the changes so that all of the checks are now passing. Could you take a look to see if we can complete the PR?

Michael

File: specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/preview/2015-06-15/Network.json [hunk 2]
```diff
@@ -75,6 +75,102 @@
...
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/providers/Microsoft.Network.Admin/locations/{location}/operations": {
+      "get": {
+        "x-ms-examples": {
+          "Returns the list of support REST operations.": {
+            "$ref": "./examples/Locations/List.json"
```
Comment (weidongxu-microsoft) [inline specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/preview/2015-06-15/Network.json]: This seems not a matching example json.

File: specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/preview/2015-06-15/Network.json [hunk 3]
```diff
@@ -119,14 +215,29 @@
           "description": "The name of the operation being performed on this particular object. This name should match the name that appears in RBAC or the event service.",
           "type": "string",
           "readOnly": true
-        },
-        "display": {
```
Comment (weidongxu-microsoft) [inline specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/preview/2015-06-15/Network.json]: Is this supposed to be a non-existing property in older spec, or is it a breaking change (no longer backward compatible)?

File: specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/preview/2015-06-15/examples/Locations/List.json [hunk 4]
```diff
@@ -0,0 +1,15 @@
+{
+  "parameters": {
+    "subscriptionId": "20ff7fc3-e762-44dd-bd96-b71116dcdc23",
+    "location": "westus",
+    "api-version": "2015-06-15"
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "value": [],
```
Comment (weidongxu-microsoft) [inline specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/preview/2015-06-15/examples/Locations/List.json]: You don't have to, but to me an empty list does not provide much insight as an example...

**Comment by weidongxu-microsoft** [comment]
@zhiyuanxia
I mainly focus on details of swagger definition and autorest extensions etc. (I am more concerned about the code generated via autorest on the API)
ARM will review the API change. That could take a couple of days.
And you seems to have a breaking change in Operation definition. (doc https://microsoft.sharepoint.com/:w:/t/azureresourcemanagerteam/EWXsAQ1yx25KkyYCeeWGUgwBSBxEdUDEbHi6FZ__U8EOQw?rtime=2ef6LVcg10g)

File: specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/preview/2015-06-15/Network.json (hunk 3 above)
Comment (zhiyuanxia) [reply specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/preview/2015-06-15/Network.json]: Good catch! This was removed by mistake. I have reverted this removal.

File: specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/preview/2015-06-15/examples/Locations/List.json (hunk 4 above)
Comment (zhiyuanxia) [reply specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/preview/2015-06-15/examples/Locations/List.json]: I took the operation example as a the template for this example. At this time, the location operation isn't really supported OnPrem. This operation is here to keep consistency with public azure.

File: specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/preview/2015-06-15/Network.json (hunk 2 above)
Comment (zhiyuanxia) [reply specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/preview/2015-06-15/Network.json]: I believe the example should be valid. Per below, the location operation isn't really supported OnPrem. This operation is here to keep consistency with public azure.

**Comment by zhiyuanxia** [comment]
Thank you for taking a look at the pull request, Weidong.

I have made some additions based on your comments and removed the breaking change. Please let me know if there is anything I can do to help the ARM team with their review.

Thanks,
Michael

File: specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/preview/2015-06-15/Network.json (hunk 2 above)
Comment (weidongxu-microsoft) [reply specification/azsadmin/resource-manager/network/Microsoft.Network.Admin/preview/2015-06-15/Network.json]: What I mean is that this /{location}/**operations** API refer to /**Locations**/List.json, while above /**locations** API refer to /**Operations**/List.json. Seems reversed to me.

Anyway, not an important issue here.

**Review by deathly809 — APPROVED** [review]
Looks good to me.

**Comment by martabarillas** [comment]
@zhiyuanxia please address my comments.

**Comment by weidongxu-microsoft** [comment]
Raised to armapireview@microsoft.com, still waiting for feedback.

**Comment by zhiyuanxia** [comment]
Hi Marta,

Can you clarify which comments you are referring to? I might be missing something but I don’t see any comments on the PR. Do you mean the codeflow comments? I have addressed them in a later iteration. I will add my new changes to the codeflow review.

Thanks,
Michael

**Comment by deathly809** [comment]
@zhiyuanxia 

> Hi Marta, Can you clarify which comments you are referring to? I might be missing something but I don’t see any comments on the PR. Do you mean the codeflow comments? I have addressed them in a later iteration. I will add my new changes to the codeflow review. Thanks, Michael 

She probably forgot to submit her review.  You can't see review comments unless the review has been submitted :(

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**